Sources/CoffeeAddictsLibrary/Helpers/CoffeeShopsCSVParser.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  File.swift
+//  
+//
+//  Created by Mihai Garda Popescu on 23.07.2021.
+//
+
+import Foundation
+import CoreLocation
+
+typealias CoffeeShopsResult = ([CoffeeShop], ParsingStatus) -> Void
+
+class CoffeeShopsCSVParser {
+    static func parseCSVString(_ string: String, completion: @escaping CoffeeShopsResult) {
+        var coffeeShops: [CoffeeShop] = []
+        
+        let parsedCSV: [[String]] = string.components(separatedBy: "\n").map{ $0.components(separatedBy: ",") }
+        
+        if parsedCSV.count == 0 {
+            completion([], .CSVFileInvalid)
+        }
+        
+        for line in parsedCSV {
+            if line.count != 3 {
+                continue
+            }
+            if let lat = Double(line[1].trimmingCharacters(in: .newlines)), let lng = Double(line[2].trimmingCharacters(in: .newlines)) {
+                if CoffeeShopsLocationHelper.locationIsValid(location: CLLocation(latitude: lat, longitude: lng)) {
```

Comment:
Can you break also this one in multiple methods?

Reply:
Partially done, I moved the lat/lng check into a separate method. I'd rather leave a for loop with several continue statements as it is, I find it more easily to follow where it will skip.